src/engine.cc

```diff
@@ -131,14 +133,15 @@ SearchLimits EngineController::PopulateSearchLimits(int ply, bool is_black,
   // How to scale moves time.
   float slowmover = options_.Get<float>(kSlowMoverStr);
   int64_t move_overhead = options_.Get<int>(kMoveOverheadStr);
+  int64_t lagbuffer = options_.Get<int>(kLagBufferStr);
   float time_curve_peak = options_.Get<float>(kTimeCurvePeak);
   float time_curve_left_width = options_.Get<float>(kTimeCurveLeftWidth);
   float time_curve_right_width = options_.Get<float>(kTimeCurveRightWidth);
 
   // Total time till control including increments.
   auto total_moves_time =
       std::max(int64_t{0},
-               time + increment * (movestogo - 1) - move_overhead * movestogo);
+               time - lagbuffer + increment * (movestogo - 1) - move_overhead * movestogo);
```

Comment:
If it is working as a "worst case" buffer then I don't think you need this change in this line.